senderBuffer.go

```diff
@@ -1,10 +1,61 @@
 package sctp
 
 type senderBuffer struct {
-	buffer                         [][]byte       // Main sender buffer needed
-	splits                         [][][]byte     // splits for buffer for each path
-	paths                          []*Association // Individual paths - can access MTU of each path from paths.mtu
-	bufferedAmount                 []uint32       // bufferredAmount[i] is the buffer size occupied by chunks on paths[i]
-	bufferedAmountOutstandingBytes []uint32       // bufferedAmountOutstandingBytes[i] is the buffer size occupied by outstanding chunks on paths[i]
-	bufferSize                     uint32         // total buffer size
+	buffer                         []*chunkPayloadData   // Main sender buffer needed
+	splits                         [][]*chunkPayloadData // splits for buffer for each path
+	paths                          []*Association        // Individual paths - can access MTU of each path from paths.mtu
+	bufferedAmount                 []uint32              // bufferredAmount[i] is the buffer size occupied by chunks on paths[i]
+	bufferedAmountOutstandingBytes []uint32              // bufferedAmountOutstandingBytes[i] is the buffer size occupied by outstanding chunks on paths[i]
+	bufferSize                     uint32                // total buffer size
+}
+
+func newSenderBuffer(paths []*Association) *senderBuffer {
+	senderBuff := senderBuffer{buffer: []*chunkPayloadData{}, paths: paths, splits: [][]*chunkPayloadData{}}
+	senderBuff.bufferSize = uint32(len(senderBuff.buffer))
+	a := make([][]*chunkPayloadData, len(paths))
+	for i := range a {
+		a[i] = make([]*chunkPayloadData, 0)
+	}
```

Comment:
```suggestion
```

I don't think these lines are necessary? `nil` slices are equivalent to slices of length zero.